### ISSUE TITLE
Statements re-entrancy.

### ISSUE THREAD

**Issue by mkanada** [square/javapoet#332]
I'm getting the exception "java.lang.IllegalStateException: statements cannot be re-entrant"
but I don't know what are the statement that is wrong.

The exception message could give some information, somewhat like a generation stacktrace.

**Comment by swankjesse** [comment]
Yeah, that's an awful message. Its slightly nicer in the latest code. The issue is you can't have two`$[` in sequence.

**Comment by swankjesse** [comment]
(Fix by using `$>$>`, or by closing your `$[`.
